pandas/core/generic.py

```diff
@@ -2792,6 +2795,9 @@ def to_hdf(
         )
 
     @final
+    @deprecate_nonkeyword_arguments(
+        version="3.0", allowed_args=["self", "name"], name="to_sql"
+    )
     def to_sql(
```

Comment:
It might make sense to allow `con` as positional as well?
(certainly given that it is a required argument, quite self-descriptive (eg not a boolean argument), and that this pattern is widely used in our own docs)

Reply:
Yeah I can see that being reasonable. I don't have super strong feelings about it though

Reply:
Let me know if I need to update the PR!

Reply:
I agree with @jorisvandenbossche, ``con`` is a weird name, I'd allow this to be positional as well (personally never specified it). Opened #54749 and PR is incoming, this should go into 2.1